### PR TITLE
crimson: add seastore and bluestore to crimson-store-nbd

### PR DESCRIPTION
**Pull request by athanatos**
<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by athanatos** [comment]
This is a somewhat more convenient way to trigger crashes in seastore.  Later on, it'll be useful for comparing workload behavior between bluestore and seastore.

File: src/crimson/tools/store_nbd/fs_driver.cc [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+	).then([this, i](auto coll) {
+	  ceph::os::Transaction t;
+	  t.create_collection(get_coll(i), 0);
+	  return fs->do_transaction(coll, std::move(t));
+	});
+      });
+  }).then([this] {
+    return fs->umount();
+  }).then([this] {
+    return fs->stop();
```
Comment (tchaikov) [inline src/crimson/tools/store_nbd/fs_driver.cc:113]: i don't think this `fs->stop()` is necessary.

File: src/crimson/tools/store_nbd/fs_driver.cc [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+	  return seastar::now();
+	});
+      });
+  });
+};
+
+seastar::future<> FSDriver::close()
+{
+  collections.clear();
+  return fs->stop(
```
Comment (tchaikov) [inline src/crimson/tools/store_nbd/fs_driver.cc:147]: i don't think this `fs->stop()` is necessary.

File: src/crimson/tools/store_nbd/fs_driver.cc [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+  ).then([this] {
+    return fs->umount();
+  }).then([this] {
+    return fs->stop();
+  }).then([this] {
+    fs.reset();
+    return fs->stat();
```
Comment (tchaikov) [inline src/crimson/tools/store_nbd/fs_driver.cc:154]: after `fs.reset()`, i think the `FuturizedStore` instance managed by `fs` is a nullptr. so the behavior of `fs->stat()` would be undefined.

Comment (athanatos) [reply src/crimson/tools/store_nbd/fs_driver.cc:154]: Yeah, that stat() is supposed to be in mount().  Will fix.

File: src/crimson/tools/store_nbd/fs_driver.h [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  size_t get_size() const {
+    return size;
+  }
+
+  seastar::future<> mount() final;
+
+  seastar::future<> close() final;
+
+private:
+  size_t size;
```
Comment (tchaikov) [inline src/crimson/tools/store_nbd/fs_driver.h:35]: `size` is initialized in `mount()`, but before that it is not. might want to set it to, for instance, `0`?

**Review by tchaikov — APPROVED** [review]

**Review by tchaikov — APPROVED** [review]

**Pull request merged**